internal/lara/lara.go

```
@@ -64,46 +79,47 @@ func MakeLara(rpc *ethclient.Client, signing_key, deployment_address, oracle_add
-			// end the epoch
-			l.Snapshot()
-			// wait 3 sec
-			time.Sleep(4 * time.Second)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-			l.Compound()
+	// Start listening for new snapshots
+	go l.ListenForSnapshots(ctx)
 
+	for {
```

golangci-lint (1.64.8)

S1000: should use for range instead of for { select {} }

(gosimple)